components/drivers/sdio/block_dev.c

```diff
@@ -689,22 +690,25 @@ void rt_mmcsd_blk_remove(struct rt_mmcsd_card *card)
     rt_list_t *l, *n;
     struct mmcsd_blk_device *blk_dev;
 
-    for (l = (&blk_devices)->next, n = l->next; l != &blk_devices; l = n, n = n->next)
+    if (card->blk_devices.next != RT_NULL)
     {
-        blk_dev = (struct mmcsd_blk_device *)rt_list_entry(l, struct mmcsd_blk_device, list);
-        if (blk_dev->card == card)
+        for (l = (&blk_devices)->next, n = l->next; l != &blk_devices; l = n, n = n->next)
         {
-            /* unmount file system */
-            const char *mounted_path = dfs_filesystem_get_mounted_path(&(blk_dev->dev));
-            if (mounted_path)
+            blk_dev = (struct mmcsd_blk_device *)rt_list_entry(l, struct mmcsd_blk_device, list);
+            if (blk_dev->card == card)
             {
-                dfs_unmount(mounted_path);
-                LOG_D("unmount file system %s for device %s.\r\n", mounted_path, blk_dev->dev.parent.name);
+                /* unmount file system */
+                const char *mounted_path = dfs_filesystem_get_mounted_path(&(blk_dev->dev));
+                if (mounted_path)
+                {
+                    dfs_unmount(mounted_path);
+                    LOG_D("unmount file system %s for device %s.\r\n", mounted_path, blk_dev->dev.parent.name);
```

Comment:
LOG_x 有自动换行功能，不需要额外再加换行符了

```suggestion
                    LOG_D("unmount file system %s for device %s.", mounted_path, blk_dev->dev.parent.name);
```